craft_archives/repo/apt_key_manager.py

```diff
@@ -267,3 +272,12 @@ def _create_keyrings_path(self):
                 f"Keyrings location {self._keyrings_path} doesn't exist; Attempting to create it."
             )
             self._keyrings_path.mkdir(mode=0o755, parents=False)
+
+
+@contextmanager
+def _temporary_home_dir() -> Iterator[pathlib.Path]:
```

Comment:
not sure if `Iterator[Path]` is the correct annotation here

Reply:
Maybe `typing.ContextManager`?

Reply:
Oh neat I tried `ContextManager` and pyright complained that the thing is actually an `Iterator`, so I guessed right after all